rhizome/kubernetes/lib/ubi_cni.rb

```diff
@@ -219,17 +219,23 @@ def error_exit(message)
     exit 1
   end
 
-  def find_random_available_ip(subnet)
-    allocated_ips = @ipam_store["allocated_ips"].values.flatten
-    max_retries = 100
+  def find_random_available_ip(subnet, reserved_ips: [])
+    allocated_ips = @ipam_store["allocated_ips"].values.flatten + reserved_ips
 
-    max_retries.times do
-      ip = generate_random_ip(subnet)
-      unless allocated_ips.include?(ip.to_s)
-        return ip
+    if subnet.ipv4?
+      available_ips = generate_all_usable_ips(subnet).reject { |ip| allocated_ips.include?(ip.to_s) }
```

Comment:
`generate_all_usable_ips` returns a new array, so `reject!` is more efficient (but you cannot chain it):

```suggestion
      available_ips = generate_all_usable_ips(subnet)
      available_ips.reject! { |ip| allocated_ips.include?(ip.to_s) }
```